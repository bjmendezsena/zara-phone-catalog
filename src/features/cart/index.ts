export * from "./context";
export * from "./helpers";
export * from "./types";
export * from "./components";
