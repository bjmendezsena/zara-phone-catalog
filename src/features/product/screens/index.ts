export * from "./ProductDetails";
export * from "./ProductGrid";

