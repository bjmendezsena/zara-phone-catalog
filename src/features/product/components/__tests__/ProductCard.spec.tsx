import { render, mockProductListResponse, getHtml } from "@/tests";
import { ProductCard } from "@/features/product";

describe("ProductCard", () => {
  const product = mockProductListResponse[0];

  it("renders product details correctly", () => {
    const { container } = render(<ProductCard product={product} />);
    expect(getHtml(container)).toMatchSnapshot();
  });

  it("has correct link to product details", () => {
    const { getByRole } = render(<ProductCard product={product} />);
    const link = getByRole("link");

    expect(link).toHaveAttribute("href", `/product/${product.id}`);
  });
});
