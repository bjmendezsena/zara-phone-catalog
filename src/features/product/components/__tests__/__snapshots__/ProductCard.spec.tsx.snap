// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ProductCard renders product details correctly 1`] = `
"<div>
  <a
    aria-label="Ver detalles del producto Samsung Galaxy S24 Ultra"
    class="w-full h-full max-h-[344px] flex items-center justify-center text-center transition-all duration-300 focus:outline-none focus:ring-2 focus:ring-black focus:ring-offset-2 border-1 flex-col relative overflow-hidden group"
    data-testid="product-card"
    href="/product/SMG-S24U"
  >
    <div
      class="absolute inset-0 bg-primary transform translate-y-full group-hover:translate-y-0 transition-transform duration-500 ease-out"
    />
    <div
      class="relative z-10 w-full h-full flex flex-col"
    >
      <div
        aria-label="Imagen del producto Samsung Galaxy S24 Ultra"
        class="flex-1 flex items-center justify-center p-4"
      >
        <div
          class="w-full max-w-[150px] h-full max-h-[200px] flex items-center justify-center"
        >
          <img
            alt="Samsung Galaxy S24 Ultra"
            class="w-full h-full object-contain group-hover:brightness-110 transition-all duration-300"
            data-priority="true"
            height="200"
            src="http://prueba-tecnica-api-tienda-moviles.onrender.com/images/SMG-S24U-titanium-violet.webp"
            width="200"
          />
        </div>
      </div>
      <div
        class="flex-shrink-0 w-full flex justify-between items-end px-4 py-3"
      >
        <div
          class="flex flex-col space-y-1 items-start"
        >
          <span
            class="text-xs text-secondary font-light group-hover:text-white transition-colors duration-300 uppercase tracking-wide"
          >
            Samsung
          </span>
          <span
            class="text-sm text-primary font-light group-hover:text-white transition-colors duration-300 leading-tight"
          >
            Galaxy S24 Ultra
          </span>
        </div>
        <span
          class="text-sm text-primary font-medium group-hover:text-white transition-colors duration-300"
        >
          1329.00 EUR
        </span>
      </div>
    </div>
  </a>
</div>"
`;
